
\chapter[Expanding the atlas of human disease variants]{Expanding the atlas of variant effects in human disease genes}

\section{Introduction}

Within coming decades, millions of people will have their genome sequenced. Unfortunately, we have limited ability to interpret personal genomes, each carrying 100-400 rare missense variants~\cite{the_1000_genomes_project_consortium_global_2015} of which many must currently be classified as Variants of Uncertain Significance (VUS). For example, gene panel sequencing aimed at identifying germline cancer risk variants in families yielded VUS for the majority of missense variants~\cite{maxwell_evaluation_2016}. While functional variants can be predicted via computational tools such as PolyPhen-2~\cite{adzhubei_predicting_2001} and PROVEAN~\cite{choi_predicting_2012}, these methods can confidently detect only one third as many disease variants as are detectable by experimental assays~\cite{sun_extended_2016}. Unfortunately, experimental assays are either unavailable or economically inviable for most human disease genes. 

Recent DMS studies have provided individual maps the critical RING domain of BRCA1~\cite{starita_massively_2015} associated with breast cancer risk, and the PPAR$\gamma$ protein associated with Mendelian lipodystrophy and increased risk of type 2 diabetes~\cite{majithia_prospective_2016}. Such maps can not only identify functionality of a clinical variant accurately, but also potentially do so in advance of that variant's first clinical presentation. 

In the previous chapter, we established a framework for comprehensive high-quality screening of the functional effects across all possible missense mutations in human genes. The functional complementation assay used in the assay allows for the generation of maps that not only represent the overall functional consequences of mutations, but also serves as a common basis to make maps more directly comparable. In addition, the statistical analysis and machine learning component we introduced supports allows for high overall map quality and completeness. Using this framework we have created a complete functional map for the SUMO E2 conjugase UBE2I. Here we create a map of a second member of the Sumoylation pathway, SUMO1. We examine both map in detail before discussing the interpretation of yeast complementation phenotypes in terms of humans. 

To demonstrate the value of our frameworks in terms of clinical interpretation of variants, we proceed to add a diverse set of six new disease gene maps to our atlas: TPK1 encoding Thiamin Pyrophosphokinase 1, NCS1 encoding Neuronal Calcium Sensor 1, as well as the paralogues CALM1, CALM2 and CALM3, which each encode the protein Calmodulin. We evaluate the maps in terms of pathogenicity prediction and VUS reclassification.

\section{Results}

% \todo{Outline the structure of the results section}

\subsection{Functional map of SUMO E2 recapitulates known biology and poses new questions}

% \todo{Quickly recap how maps for UBE2I and SUMO1 were made}

The DMS map of UBE2I produced in the previous chapter paints a comprehensive picture of variant effects on protein function (Figure~\ref{fig:ube2i-map}). Based on the map, several observations can be made. Consistent with the results of smaller-scale biochemical studies of the SUMO E2 conjugase\cite{bencsath_identification_2002,bernier-villamor_structural_2002}, the areas most sensitive to mutation are those proximal to the active site (particularly residues 81-88, 90, 92-96, and 127-130), and the N-terminal $\alpha$-helix which mediates four protein interactions including the critical interaction with the E1 SUMO-activating complex. Within the active site, particularly strong sensitivity to mutation can be observed at the Cystein residue at position 93. This is consistent with its central role in E2 function, as it forms a thioester bond with the SUMO C-terminus~\cite{bernier-villamor_structural_2002}.

An interesting feature of the map is the alternating tendency towards damaging and benign substitutions across positions 55-65. A comparison with solvent accessibility reveals this to be caused by alternating externally and internally-oriented residues, with the latter positions constrained to be hydrophobic. This alternating tendency is also reflected in evolutionary conservation across these positions. 

All protein-protein interaction interfaces previously captured in co-crystal show increased sensitivity to mutation when compared to other surface residues (Figure~\ref{fig:ube2i_interfaces}). When comparing individual protein interaction interfaces, the most substantial fitness defects are observed in those for the E1 activating complex binding interface and the covalent and noncovalent SUMO binding interfaces (Figure~\ref{fig:ube2i_interfaces}A). While the homodimerization interface also shows significant sensitivity, the effects are not as severe as those at the E1 interface (Figure~\ref{fig:ube2i_interfaces}B). This is consitent with the Alontaga and colleagues' hypothesis regarding its involvement in SUMO chaining~\cite{alontaga_rwd_2015}, as in yeast SUMO chain formation has so far only been observed to be involved in meiosis, which is not a mechanism vital to fitness in a complementation assay. Alontaga \etal also postulate however, that noncovalent SUMO binding is necessary for SUMO chain formation. In contrast to the homodimerization interface, the noncovalent SUMO binding interface shows a very strong sensitivity to mutation. This may be due to two different reasons: (i) there is a 27\% overlap between the interface for noncovalent SUMO binding interface and the interface for E1-E2 binding, which is among the most sensitive surfaces in UBE2I; and (ii) noncovalent SUMO binding also plays an important role as an adapter for many E3 proteins~\cite{cappadocia_structural_2015}.
%FIXME: Cappadocia: Structural basis for catalytic activation by the human ZNF451 SUMO E3 ligase


\begin{figure}[h!]
	\centering
	\includegraphics[width=\textwidth]{img/ube2i_interfaces.pdf}
	\caption{Complementation fitness of mutations at interaction interfaces. A) Median mutant fitness mapped to the crystal structure of UBE2I. The $\Psi$KXE substrate recognition motif is shown as green stick model, covalently and non-covalently bound SUMO are shown as crimson and brown cartoon model, respectively. B) Mutant fitness scores distributions for residues at different interaction interfaces. }
	\label{fig:ube2i_interfaces}
\end{figure}

Another interesting observation can be made with respect to a known phosphorylation site on the surface of UBE2I. Su and colleagues previously discovered that phosphorylation of Serine 71 via the Cyclin-dependent Kinase CDK1 results in sumoylation hyperactivity~\cite{su_phosphorylaton_2012}. Our map shows that substitutions with phosphomimetic residues at this position lead to hyperactive complementation, consistent with Su \etal.'s observations. Furthermore, other residues amenable to phosphorylation are also tolerated, while hydrophobic replacements are generally deleterious (Figure~\ref{fig:phosphosite}).
%FIXME: Su2012: Phosphorylation of Ubc9 by Cdk1 enhances SUMOylation activity

\begin{figure}
	\centering
	\includegraphics[width=\textwidth]{img/phosphosite.pdf}
	\caption{Phoshporylation site of UBE2I shows hyperactive complementation when mutated to phosphomimetic residues.}
	\label{fig:phosphosite}
\end{figure}


\subsubsection{Substrate specificity shifts and E2 hyperactivity}

Intriguingly, many sites show fitness that is better than wildtype (e.g., positions 74, 76, 88, 89, 91 and 98). Manual functional complementation spotting assays confirmed that complementation with these mutants allows greater growth than does the wild type human protein, but resemble more closely the growth at the permissive temperature for the \textit{ubc9-ts} strain (Figure~\ref{fig:hyperactive}A). One might be tempted to interpret these cases as reversions to residues present in the yeast protein. However, a comparison of fitness score distributions between changes to S. cerevisiae  residues and those occurring in the distant species \species{Dictyostelium~discoideum} (amoeba) or \species{Drosophila~melanogaster} (fly) showed no significant difference (Figure~\ref{fig:hyperactive}B). Recognizing that in this assay, human UBE2I must function with the yeast versions of other sumoylation pathway members, it stands to reason that some substitutions could be adaptive by improving compatibility with yeast interaction partners. A comparison with co-crystal structure data~\cite{gareau_determinants_2012} shows that many of the apparently-adaptive residues are located on the surface facing the general direction of the substrate, with some being in direct contact with the substrate's sumoylation motif (Figure~\ref{fig:hyperactive}C). This suggests a possible adaptation via improved recognition of substrates for which sumoylation is most important for yeast growth. Indeed, \textit{in vitro} sumoylation assays performed previously for a small number of UBE2I mutants revealed increased sumoylation for some substrates~\cite{bernier-villamor_structural_2002}. Comparing our map with these sumoylation assay results, we confirmed that above-WT complementation levels were enriched for cases of substrate specificity shift (Figure~\ref{fig:hyperactive}D). \todo{This needs to be discussed in more detail}


\begin{figure}[h!]
	\centering
	\includegraphics[width=\textwidth]{img/hyperactive.pdf}
	\caption{Hyperactive complementation in UBE2I. A) Variants scoring higher than the wildtype controls show stronger growth in manual complementation spotting assays and resemble the WT yeast. B) Distribution of scores for changes to residues naturally occurring in yeast, amoeba and fly are not significantly different from each other. C) Maximum mutant score mapped to amino acid positions on UBE2I structure. Hyperactive mutations are clustered at the substrate recognition site. D) \textit{In vitro} sumoylation assays performed by Bernier-Villamor~\etal~\cite{bernier-villamor_structural_2002} in comparison to complementation fitness scores.}
	\label{fig:hyperactive}
\end{figure}


As Figure~\ref{hyperactive}D shows, substrate specificity does not paint a complete picture of the mechanisms potentially underlying hyperactive complementation. A particularly interesting exception can be observed at residues A15 and T108. Both residues harbor hyperactive mutations but do not face towards the substrate. Instead, they form part of the interface with the E3 SUMO ligase RanBP2, and flank a small cavity on UBE2I's surface into which RanBP2 inserts a phenylalanine residue upon binding~\cite{gareau_determinants_2012}. Changing either A15 or T108 into aromatic residues results in a large fitness increase (Figure~\ref{fig:pi-stack}). This may be the result from the emergence of a $\pi$-stack interaction that strengthens E2-E3 binding.

\begin{figure}[h!]
	\centering
	\includegraphics[width=\textwidth]{img/pi-stack.pdf}
	\caption{Potential de-novo pi-stack interaction between UBE2I and the E3 RanBP2}
	\label{fig:pi-stack}
\end{figure}

It is unclear how to interpret the effect of mutations that enhance growth in the yeast complementation assay. If fitness measured in the assay is directly proportional to fitness in the real biological context, then these enhancing mutations would be beneficial. However one can also imagine an alternative scenario in which activity-enhancing mutations are deleterious in the real biological context. To objectively distinguish between these possibilities, we collaborated with Jesse Bloom to employ a method he recently published that leverages likelihood-based phylogenetics to quantitatively compare how well different experimental measurements represent actual evolutionary constraints in nature~\cite{bloom_experimentally_2014,bloom_identification_2017}. We compared three models relating the experimental fitness to the evolutionary preference for a mutated amino-acid sequence: (a) the evolutionary preference was directly proportional to the untransformed experimental fitness; (b) the preference had a ceiling at the wildtype experimental fitness (values greater than 1 were set to 1); or (c) the preference was set to the reciprocal of fitness for mutations with greater-than-wildtype scores, corresponding to a deleterious effect of enhancing mutations. Dr. Bloom kindly provided the phydms software~\cite{bloom_identification_2017} to test which of these three approaches best described the evolutionary constraint on a set of naturally occurring UBE2I homologs, using fitness scores that excluded conservation features from the regularization process, to avoid the circularity of using natural sequence data when deriving the scores. As shown in Table~\ref{tab:phydms}, the best fit is achieved using the model that assumes that enhancing mutations are deleterious. This result provides objective support for the idea that mutations that enhance activity above wildtype levels in the complementation assay are actually deleterious in a real biological context.

Based on these observations we reinterpreted cases of hyperactive complementation in our map as deleterious and repeated the imputation and regularization procedure. This lowered the cross-validation RMSD substantially from $0.33$ to $0.24$.

\begin{table}[h!]
	\centering
	\caption{Comparison of different models for the effects of hyperactivating mutations. AIC: Akaike Information Criterion\newline}
	\begin{tabular}{l l}
Model & $\Delta$AIC relative to best model\\ \hline\hline
Penalize hyperactive mutations & 0\\
Cap score of hyperactive mutations at wildtype level & 27.7\\
Hyperactive mutations as beneficial	& 60.6	
	\end{tabular}
	\label{tab:phydms}
\end{table}



\subsubsection{Intragenic epistasis and compensatory mutations}

Full-length UBE2I clones generated for DMS-BarSEQ analysis often encoded more than one amino acid change. Multi-mutant clones offer the opportunity to search for intragenic genetic interactions. Genetic interaction is defined when a combination of mutations yields an unexpected phenotypic effect, so that identifying genetic interactions requires that we model the phenotype expected from a combination of mutations, given the single-mutant effects.  Here we used a previously-described multiplicative model 20,21 in which genetic interaction is measured as $\varepsilon_{ij} = f_i \cdot f_j - f_{ij}$, where $f_i$ and $f_j$ represent single mutant fitness and fij represents double mutant fitness scores. Most double mutants (71\%) did not show a significant deviation from $\varepsilon_{ij} = 0$ under this model, while 328 position pairs did show significant genetic interaction (Figure~\ref{fig:epsistasis}, see Methods). Of particular interest are compensatory interactions, i.e. cases where a double mutation is more fit than either of the component single mutations.  Where compensatory residues are proximal in the protein structure, the combination of two mutant residues may be able to re-establish a physical interaction that was lost in each of the single mutants. Although the majority of genetically interacting sites were not proximal in the structure (Figure~\ref{fig:epsistasis}B), there were interesting exceptions. For example, the I4T-P69S double mutant appears to exhibit compensatory behaviour: In the wild type structure, the van-der-Waals radii of the two residues are in direct contact (Figure~\ref{fig:epsistasis}C). Either mutation alone would be expected to destabilize the hydrophobic interaction between isoleucine and proline.  However, In the double mutant, hydroxyl groups on the two residues could adopt a hydrogen bond that re-establishes interaction and re-stabilizes the fold (Figure~\ref{fig:epsistasis}D).

\begin{figure}[h!]
	\centering
	\includegraphics[width=\textwidth]{img/epistasis.pdf}
	\caption{Epistasis}
	\label{fig:epsistasis}
\end{figure}


\subsection{A functional map for SUMO1}

Using the DMS-TileSeq version of the framework established in the previous chapter we also created a complete functional map for SUMO1 (Figure~\ref{fig:sumo-map}A). The most immediately apparent feature of the SUMO1 map was the strong enrichment for neutral substitutions within the first 20 amino acid positions, which is consistent both with the low level of evolutionary conservation for this region and its annotation as a disordered region. The last four amino acid positions appeared similarly insensitive to mutation, consistent with the cleavage of this region by SENP proteases during SUMO maturation. By contrast, other residue positions were strongly sensitive to mutation, including many inward-facing residues that are apparently constrained to be hydrophobic. As expected, the C-terminal diglycine, directly preceeding the last four cleaved residues, is also very sensitive to mutation, as it is required for the covalent binding of SUMO to the E1, the E2 and to the sumoylation target protein.Interestingly, except for the C-terminal diglycine, the residues that directly touch the E2 during covalent binding are not as sensitive (Figure\ref{fig:sumo-map}B). This may be due to  SUMO being force-fed to the E2 by the E1 activating complex and the thioester bond it forms with the E2's cystein \#93 is all that is needed to maintain the complex. By contrast, residues in the interface for non-covalent E2 binding are much more sensitive (Figure\ref{fig:sumo-map}C). Especially leucine \#80 and methionine \#82 appear to be important

Other strongly constrained residues are core members of interaction interfaces. These include the central phenylalanine 36 in the SUMO recognition motif (SRM) interface; glycine 68, which forms the apex of a tight turn within the interface with de-sumoylation enzymes, as well as the E1 and E2 proteins; and leucine 80, which is part of the interface with non-covalently bound E2. 

\begin{figure}[h!]
	\centering
	\includegraphics[width=\textwidth]{img/sumo-map.pdf}
	\caption{Functional map of SUMO1}
	\label{fig:sumo-map}
\end{figure}

The proximity and orientation of aspartate \#73 and lysine \#48 suggests that they are able to form a salt bridge with one another.  The importance of each residue according to the DMS map supports a model in which this salt bridge is important for SUMO folding and/or stability. Interestingly, substituting aspartate for methionine \#59, which points towards lysine \#48 from an angle similar to that of aspartate \#73, enhances the complementation fitness of SUMO1 beyond wild type levels.  This further underlines the potential importance of a polar interaction involving lysine \#48 (Figure~\ref{fig:saltbridge}).

\begin{figure}[h!]
	\centering
	\includegraphics[width=\textwidth]{img/saltbridge.pdf}
	\caption{A salt bridge within SUMO1 between Asp73 and Lys38 appears important for stability. Met59Asp may increase stability even further.}
	\label{fig:saltbridge}
\end{figure}


% As was shown above for UBE2I, phylogenetic analysis of SUMO1 similarly showed that adaptive mutations with ability to complement yeast better than wild-type are likely deleterious in humans. We therefore transformed fitness scores so that such adaptive mutations are considered to be deleterious (see Methods).  However, because adaptive substitutions may provide interesting clues about differences between yeast and human cellular contexts, we provide both transformed (Figure 5) and untransformed versions of each map.




\subsection{Functional maps of three human disease genes}

Having established and evaluated our Deep Mutational Scanning framework on two members of the sumoylation pathway, we aimed to create maps for a diverse set of genes that have been associated with disease with varying degrees of confidence. While  heterozygous null mutations in SUMO1 have previously been associated with cleft palate, we wished to create maps that could be tested in the context of variant classification in terms of disease. Based on the availability of robust complementation assays, we applied DMS-TileSeq to the following protein targets: Thiamine Pyrophosphokinase 1 (TPK1), associated with vitamin B1 metabolism dysfunction; Neuronal Calcium Sensor 1 (NCS1), which has been implicated in autism based on a single \textit{de novo} mutation;  and CALM1, CALM2 and CALM3 associated with heart conditions long-QT syndrome and catecholaminergic polymorphic ventricular tachycardia. Although the three calmodulin genes differ in nucleotide sequence, each encodes the same polypeptide sequence. Thus, we performed a deep mutational scan only for CALM1, which enabled us to also map missense variant effects in CALM2 and CALM3. In each case, we used the TileSEQ approach coupled with complementation to generate a map of missense variant functions. 

As was shown above for UBE2I, phylogenetic analysis of SUMO1 similarly showed that variants with ability to complement yeast better than wild-type are likely deleterious in humans. We therefore transformed fitness scores so that such adaptive mutations are considered to be deleterious \todo{expand on this}. The transformed disease gene maps can be seen in Figures \ref{fig:tpk1-map} and \ref{fig:calm+ncs1-maps}).  However, because adaptive substitutions may provide interesting clues about differences between yeast and human cellular contexts, we also provide untransformed versions of each map (see Appendix).

%%FIGURE: Map for TPK1/NCS1/CALM1


\subsubsection{A thiamine pyrophosphokinase map reflects a recessive phenotype}

Thiamine pyrophosphokinase (TPK1) is a protein that forms a dimer to perform its biochemical function. Its substrate, thiamine diphosphate, is bound within two active sites formed by the dimerization interface\cite{timm_crystal_2001}. That is, each monomer contributes half of the residues making up each of the two active sites.  Each monomer in turn is made up of an N-terminal globular domain and a C-terminal $\beta$-sandwich domain (Figure~\ref{fig:tpk1_structure}A). The residues most sensitive to mutation in the protein make up the hydrophobic cores of the two domains: L21, V22, W36, G48, Y53, P65, G70, Y83, L108, I122, T124, and G127 for the N-terminal domain; and L161, G168, G199, L200, V227, V229, L236, and W237 for the C-terminal domain (Figure~\ref{fig:tpk1_structure}B). As might have been expected, mutation-sensitive residues include those closely involved in forming the active sites: D46, G70, D71, D73, D100, and K103 in the N-terminal half of the active site , contacting the diphosphate portion of the substrate (Figure~\ref{fig:tpk1_structure}C). In the C-terminal half of the active site, K203, L209, G212, L214, S216, T217, and N219 show similar sensitivity. Interestingly, the tryptophan residue at position 202 appears to be insensitive to mutation despite its close and extensive contact with the thiamine ligand. By contrast, a neighbouring lysine at position 201 is surprisingly sensitive suggesting potential importance in coordinating the ligand.  The remainder of the dimerization interface also features a number of sensitive residues, such as M136, G184, V188, G189 and G211. Finally, residues 1-12, which form a $\beta$-strand anchoring the N-terminal domain back to the C-terminal domain were also found to be sensitive.

\begin{landscape}
\begin{figure}[h!]
	\centering
	\includegraphics[width=9in]{img/tpk1-map.pdf}
	\caption{Functional map of Thiamine pyrophosphokinase 1 (TPK1). From top to bottom: Position-wise evolutionary conservation (AMAS); Secondary structure; Relative solvent accessibililty; Relative burial in homodimerization interface; Contacts with Thiamine diphosphate; Summary track showing the shares of amino acid changes resulting in varying degrees of fitness effects; Detailed heatmap showing individual amino acid change effects.}
	\label{fig:tpk1_structure}
\end{figure}
\end{landscape}

\begin{figure}[h!]
	\centering
	\includegraphics[width=.8\textwidth]{img/tpk1_structure.pdf}
	\caption{Thiamine pyrophosphokinase 1 colored by median complementation score. A) TPK1 homodimer structure showing one monomer as surface model, the other monomer as cartoon model. B) Hydrophobic residues facing the inside of the C-terminal $\beta$-sandwich domain are sensitive to mutation. C) Active site residues in contact with the substrate are sensitive to mutation.}
	\label{fig:tpk1-map}
\end{figure}


\subsubsection{The two calcium sensors NCS1 and Calmodulin show different profiles}

Calmodulin (CALM1/2/3) and the Neuronal Calcium Sensor protein (NCS1) are homologs (E-value $4 \cdot 10^{-5}$ when searched against the human proteome~\cite{altschul_basic_1990,the_uniprot_consortium_uniprot:_2015}) with 24\% sequence identity and 48.5\% similarity~\cite{rice_emboss:_2000}. However, they display different impact patterns despite their similar domain structure and similar molecular roles as calcium sensing proteins. Both are comprised of four Calcium-binding EF-hands, with NCS1 containing additional sequences upstream and downstream of the four hands. A comparison of previously published NMR structures reveals that the overall folds of the two proteins differ substantially~\cite{sarhan_crystallographic_2012,heidarsson_c-terminal_2012}. Calmodulin features a long central helix that separates two globular domains, called the N-lobe and the C-lobe, each comprised of two EF hands. A hydrophobic pocket serving as a binding interface for interacting proteins is nested within the C-terminal domain. NCS1, by contrast, forms a single shell-like shape, centered around a large hydrophobic crevice. This crevice acts as a binding interface for interacting proteins. Thus, the divergent DMS profiles we observed for CALM1/2/3 and NCS1 are consistent with these substantial structural differences.

The Neuronal Calcium Sensor NCS1 displays the greatest sensitivity to mutation within the N-terminal region containing the myristoylation site.  This myristoylation site is essential for anchoring NCS1 into the plasma membrane. One other residue that stands out is the tryptophan at position 30, which results in complete loss of function when replaced with any other amino acid. Like most other sensitive residues W30 is found among those contributing to the hydrophobic crevice acting as an interaction interface. Other cases include F55, F56, A104, M121, I152, and A182. An interesting observation can be made with respect to the two helices that separate the two N-terminal EF hands from the two C-terminal EF hands. A kink between the two helices brings them into an angle that allows a the globular shape of the overall protein to form. Without this kink it is conceivable that NCS1's fold would much more resemble that of Calmodulin. A glycine residue (G95) is likely responsible for forming that kink due to its helix breaking properties. This residue is also found to be quite sensitive to mutation.

Within Calmodulin, the regions most sensitive to mutation are: 1) the hydrophobic cores of the two globular domains; 2) interfacial residues for protein-protein interactions, and 3) a subset of the negatively charged residues in EF hands that contact Ca$^{2+}$ ions. Within the hydrophobic cores of the two lobes, five mutually interacting phenylalanine residues at positions 17, 69, 90, 93, and 142 stand out in particular, as all of them are found in the top 9 most sensitive residues on the map. Within the interaction interface, the residues D85, A89, F93, M100, L106, V109, L113, G114, L117, M125, V137, F142, M145, M146 are the most strongly sensitive to mutation. Regarding the four Calcium-binding EF-hand loops, we found it interesting that only a subset of the negatively-charged residues contacting Ca$^{2+}$ are even moderately sensitive. Within EF1, only D25 appears to be important, in EF2 only N61, in EF3 only D94 and D96, and in EF4 only D130 and D134. Overall, the EF3/4 in the C-lobe also appear to be more important than their N-lobe counterparts. This is in agreement with previous observations made by Sarhan and colleagues~\cite{sarhan_crystallographic_2012}, who described the C-lobe as the primary sensory mechanism. A number of unexplained sensitivities exist as well: Arginines at positions 54 and 91 show strong phenotypes despite extending from seemingly unused surfaces of the protein, offering the possibility that these residues are functionally relevant sites of interaction or modification.


\begin{landscape}
\begin{figure}[h!]
	\centering
	\includegraphics[width=9in]{img/calm+ncs1-maps.pdf}
	\caption{Functional map of Thiamine pyrophosphokinase 1 (TPK1). From top to bottom: Position-wise evolutionary conservation (AMAS); Secondary structure; Relative solvent accessibililty; Relative burial in homodimerization interface; Contacts with Thiamine diphosphate; Summary track showing the shares of amino acid changes resulting in varying degrees of fitness effects; Detailed heatmap showing individual amino acid change effects.}
	\label{fig:calm+ncs1-maps}
\end{figure}
\end{landscape}


\subsubsection{Functional maps recapitulate known disease cases}

To validate the utility of our maps in the context of human disease, we extracted known disease-associated variants from Clinvar~\cite{landrum_clinvar:_2016}, as well as rare and common polymorphisms observed independent of disease from GnomAD~\cite{lek_analysis_2016}, and somatic variants previously observed in tumors from COSMIC~\cite{forbes_cosmic:_2001}. 

For TPK1, a large number of very rare variants (minor allele frequency or $\text{MAF} < 10^{-6}$) is known from GnomAD. At first look, it appears the majority of these variants are shown to be deleterious (Figure~\ref{fig:tpk1_diploid}). This seems unlikely, given that Thiamine Metabolism Dysfunction Syndrome, reported to be caused by mutations in this gene, is a very severe disease to which patients succumb in childhood~\cite{mayr_thiamine_2011}., and that GnomAD attempts to filter out subjects with severe pediatric disease. However, the disease is also known to follow a recessive inheritance pattern, with only homozygous or compound heterozygous individuals being affected. We thus used phased sequence data from the 1000 Genomes Project~\cite{the_1000_genomes_project_consortium_global_2015} to determine the diploid genotypes in the TPK1 locus for all listed individuals and based our phenotype predictions based on the maximum fitness score of either allele. This improved prediction performance markedly, leading to complete separation between disease and non-disease genotypes. Both PROVEAN and PolyPhen-2 were also able to perfectly separate the two groups (Figure~\ref{fig:tpk1_diploid}B), so that additional compound heterozygotes with known disease status will be required to determine whether this DMS map is more useful than computational methods for classifying pathogenic TPK1 variants. 

\begin{figure}[h!]
	\centering
	\includegraphics[width=\textwidth]{img/diploid.pdf}
	\caption{TPK1 variants}
	\label{fig:tpk1_diploid}
\end{figure}


While NCS1 does not have any entries in ClinVar, a previous publication identified the variant R102Q as a \textit{de novo} variant in a single patient with autism spectrum disorder~\cite{handley_structural_2010}. While the variant did not affect overall protein folding and localization, the authors did observe that the dynamics of cytosol-membrane cycling were altered. Our complementation map did not show any functional impact for this variant.

As for NCS1, no disease-associated missense alleles are known for UBE2I and SUMO1 in ClinVar. However, a number of somatic mutations for all three genes have been observed in cancer according to COSMIC. While these can be expected to passenger mutations, one may still hypothesize that somatic variants are likely not subject to the same selection pressures as germline variants, as interference with developmental processes is not necessarily detrimental to a tumour. We thus tested whether germline polymorphisms in these three genes were enriched for being functional compared to their somatic counterparts in our maps. Indeed, we observed a significant difference between the two sets (Wilcoxon $P = 2.6 \cdot 10^{-5}$) (Figure~\ref{fig:calm_disease}C).

Finally, we examined our functional map of Calmodulin and found that it was able to distinguish disease variants from non-disease variants very well (Figure~\ref{fig:calm_disease}A). In contrast to TPK1, the Calmodulin map did not need to be corrected for diploid genotypes, as previously reported disease variants have been described as following a dominant inheritance pattern~\cite{crotti_calmodulin_2013}. A precision-recall (PRC) plot reveals a superior performance (AUC = 0.74) compared to PROVEAN (AUC = 0.47) and PolyPhen-2 (AUC = 0.47).  

\begin{figure}[h!]
	\centering
	\includegraphics[width=\textwidth]{img/calm_disease.pdf}
	\caption{Calmodulin variants}
	\label{fig:calm_disease}
\end{figure}


To further put our map to the test in a clinical scenario we inquired with Invitae, a company offering gene panel sequencing services for Long QT syndrome, including CALM1/2/3. In a blind test, we requested a list of Calmodulin variants they observed in patients but were unable to classify. After calibrating our map with respect to the above ClinVar and GnomAD datasets, we classified these 10 new variants (Table~\ref{tab:invitae}). Two were classified as damaging, six as benign, and two were too close to the threshold to be called either. In the next phase, Invitae revealed the associated patient cardiovascular phenotypes. Five out of the six patients with benign predictions were revealed to be healthy, while both patients with damaging predictions did show a positive phenotype. The two uncertain cases were revealed to be affected as well. A Mann-Whitney-U test showed these results to be statistically significant (P = 0.008).

% \begin{table}
% 	\centering
% 	\caption{}
% 	\begin{tabular}{l l l}
		
% 	\end{tabular}
% 	\label{tab:invitae}
% \end{table}

\section{Discussion}

\section{Methods}


